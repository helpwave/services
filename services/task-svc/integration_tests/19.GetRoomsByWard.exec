#!/bin/bash

echo "Exists, has rooms:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"ward_id": "51aca99b-2372-40b4-99ed-355dad6c014d"}' \
	localhost:3001 \
	services.task_svc.v1.RoomService/GetRoomsByWard

echo "Exists, has no rooms:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"ward_id": "626e8c70-dc93-44fb-a9d5-0b828a0a0c4b"}' \
	localhost:3001 \
	services.task_svc.v1.RoomService/GetRoomsByWard

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"ward_id": "726e8c70-dc93-44fb-a9d5-0b828a0a0c4b"}' \
	localhost:3001 \
	services.task_svc.v1.RoomService/GetRoomsByWard 2>&1 || echo ""
