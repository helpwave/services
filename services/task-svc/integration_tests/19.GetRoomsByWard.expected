Exists, has rooms:
{
  "rooms": [
    {
      "id": "80d49a5a-573e-4a7a-a9b4-f8d2431f6f14",
      "name": "Legal",
      "beds": [
        {
          "id": "1d8fbe74-37be-4ec1-beb2-0518b51e1da8",
          "name": "Turquoise"
        }
      ],
      "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406",
      "wardId": "51aca99b-2372-40b4-99ed-355dad6c014d"
    },
    {
      "id": "c7e6c19e-df2e-4130-b8f3-97b551f911ed",
      "name": "Research and Development",
      "beds": [
        {
          "id": "8612b526-9f2b-486f-9e6c-d1581ef5b6be",
          "name": "Fuscia"
        },
        {
          "id": "018bd012-4aa1-468f-b57c-c0028507c332",
          "name": "Mauv"
        },
        {
          "id": "371e0bb7-1d55-46bb-a972-4bea0df958da",
          "name": "Orange"
        },
        {
          "id": "9f5d0d44-066e-4520-aec3-46a51090d15a",
          "name": "Red"
        },
        {
          "id": "01e91d0a-88ac-402b-932e-bae326a7c234",
          "name": "Teal"
        },
        {
          "id": "c9cb785f-4db3-42f8-935a-b5c88652231b",
          "name": "Violet"
        },
        {
          "id": "99205989-53a4-4362-8280-c746d30ebc12",
          "name": "Yellow"
        }
      ],
      "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406",
      "wardId": "51aca99b-2372-40b4-99ed-355dad6c014d"
    }
  ]
}
Exists, has no rooms:
{}
Does not exist:
{}
